packages/cli/src/modules/provisioning.ee/provisioning.service.ee.ts

```diff
@@ -106,33 +106,51 @@ export class ProvisioningService {
 	}
 
 	/**
-	 * @param projectIdToRole expected to be a JSON string for a map of project IDs to role slugs
+	 * @param projectIdToRole expected to be an array of strings like this:
+	 * [
+	 *   "<projectUuid>:<projectRoleDisplayName>",
+	 *   "<projectUuid>:<projectRoleDisplayName>",
+	 *   ...
+	 * ]
 	 */
-	async provisionProjectRolesForUser(userId: string, projectIdToRole: unknown): Promise<void> {
-		if (typeof projectIdToRole !== 'string') {
+	async provisionProjectRolesForUser(userId: string, projectIdToRoles: unknown): Promise<void> {
+		if (!Array.isArray(projectIdToRoles)) {
 			this.logger.warn(
-				`Skipping project role provisioning. Invalid projectIdToRole type: expected string, received ${typeof projectIdToRole}`,
-				{ userId, projectIdToRole },
+				`Skipping project role provisioning. Invalid projectIdToRole type: expected array, received ${typeof projectIdToRoles}`,
+				{ userId, projectIdToRoles },
 			);
 			return;
 		}
 
 		let projectRoleMap: Record<string, string>;
 		try {
-			projectRoleMap = JSON.parse(projectIdToRole);
-			for (const [key, value] of Object.entries(projectRoleMap)) {
-				if (typeof key !== 'string' || typeof value !== 'string') {
+			projectRoleMap = {};
+			for (const entry of projectIdToRoles) {
+				if (typeof entry !== 'string') {
 					this.logger.warn(
-						`Skipping project role mapping for ${key}:${value}. Invalid types: expected both key and value to be strings.`,
-						{ userId },
+						`Skipping invalid project role mapping entry. Expected string, received ${typeof entry}.`,
+						{ userId, entry },
 					);
-					delete projectRoleMap[key];
+					continue;
 				}
+
+				const [projectId, roleSlugSuffix] = entry.split(':');
+				if (!projectId || !roleSlugSuffix) {
+					this.logger.warn(
+						`Skipping invalid project role mapping entry. Expected format "projectId:displayName", received "${entry}".`,
+						{ userId, entry },
+					);
+					continue;
+				}
+
+				// This means that if the external SSO setup has two roles configured
+				// for the same projectId, we only provision the one we see last in the sent list.
+				projectRoleMap[projectId] = `project:${roleSlugSuffix}`;
```

Comment:
I don't like this "manual construction of the project slug" here that much.
It can lead to undetected regressions if we ever change the way we store the slug in the project roles table.
But it works for now 👍